File: frontend/src/services/UserService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/users';

const UserService = {
    getAllUsers: () => {
        return axios.get(API_URL);
    },

    getUserById: (id) => {
        return axios.get(`${API_URL}/${id}`);
    },

    createUser: (user) => {
        return axios.post(API_URL, user);
    },

    updateUser: (id, user) => {
        return axios.put(`${API_URL}/${id}`, user);
    },

    deleteUser: (id) => {
        return axios.delete(`${API_URL}/${id}`);
    }
};

export default UserService;

File: src/main/java/com/riffo/users/service/UserService.java
package com.riffo.users.service;

import com.riffo.users.entity.User;
import com.riffo.users.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User createUser(User user) {
        return userRepository.save(user);
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }

    public void deleteUser(Long id) {
        userRepository.deleteById(id);
    }

    public User updateUser(Long id, User userDetails) {
        return userRepository.findById(id).map(user -> {
            user.setNom(userDetails.getNom());
            user.setEmail(userDetails.getEmail());
            user.setAge(userDetails.getAge());
            return userRepository.save(user);
        }).orElse(null);
    }
}
